frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -25,33 +25,62 @@ class CheckoutController < Spree::StoreController
 
     # Updates the order and advances to the next state (when possible.)
     def update
-      if OrderUpdateAttributes.new(@order, update_params, request_env: request.headers.env).apply
-        @order.temporary_address = !params[:save_user_address]
-        success = if @order.state == 'confirm'
-          @order.complete
-        else
-          @order.next
-        end
-        if !success
-          flash[:error] = @order.errors.full_messages.join("\n")
-          redirect_to(checkout_state_path(@order.state)) && return
-        end
+      if update_order
+
+        assign_temp_address
+
+        redirect_on_failure and return unless transition_forward!
```

Comment:
I find this line tough to parse for a couple reasons.

I prefer using the conditional suffix only on functions that don't mutate state. It's kind of sneaky that `transition_forward!`, the main purpose of this whole action, is at the end of the line. It also makes the order these run in mildly confusing: `this_second and this_third unless this_first`.


I'd prefer that the `transition_forward!` was more obvious, and that the order would read top-to-bottom

``` ruby
unless transition_forward!
  redirect_on_failure
  return
end
```